src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let curentState = { ...state };
+  const stateArr = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
```

Comment:
It's a good practice to validate the structure of the `actions` array to ensure that each action has the expected properties (`type`, `extraData`, `keysToRemove`). This can prevent runtime errors if the actions are not well-formed.